corehq/apps/locations/static/locations/js/location_types.js

```diff
@@ -300,8 +300,11 @@ hqDefine('locations/js/location_types.js', function(){
             if (self.expand_from() !== ROOT_LOCATION_ID){
                 if (typeof(self.expand_from()) === 'undefined' || self.expand_from() === null){
                     options = self.parents().reverse();
+                    options.push(self);
                 } else {
-                    options = self.view.loc_types_by_id()[self.expand_from()].parents();
+                    var expand_from_type = self.view.loc_types_by_id()[self.expand_from()];
+                    options = expand_from_type.parents();
+                    options.push(expand_from_type);
```

Comment:
I think you can extend your same logic beyond the level of "expand_from_type".  Say you've got the setup in these tests:
```python
    location_type_names = ['state', 'county', 'city']
    location_structure = [
        ('Massachusetts', [
            ('Middlesex', [
                ('Cambridge', []),
                ('Somerville', []),
            ]),
            ('Suffolk', [
                ('Boston', []),
                ('Revere', []),
            ])
        ]),
        ('New York', [
            ('New York City', [
                ('Manhattan', []),
                ('Brooklyn', []),
                ('Queens', []),
            ]),
        ]),
    ]
```
You want a user in Cambridge to get all the cities in Massachusetts, and all counties project-wide.  In that case, `expand_from` would be `'state'`, and `include_without_expanding` would be `county`.  The way it's done here, you can't select an `include_without_expanding` that's below your `expand_from`.